### PR TITLE
add code coverage

### PR DESCRIPTION
**Pull request by andy-neuma**
SUMMARY:
* update test runner skip to generate code coverage reports.
* add action to generate aggregate report.
* add steps to upload the aggregated report as a run artifact.
* add a test skip list defined by ".txt" files in `neuralmagic/tests`. folks can adjust the contents or override this file in automation entirely.
* make the skip list a parameter for callable workflows.
* update "remote push" workflow to use `neuralmagic/tests/skip-for-remote-push.txt`.

NOTES:
we are using the `coverage` package to generate the reports. the reports are being upload and stored in GHA artifact store. right now to view the reports they need to be downloaded and unzipped. the folder though contains everything needed to browse. eventually, we'll need to properly host the reports and add links to the run summary for quick inspection.

the expectation is that folks will iterate and converge to a desired "skip list". the one that is part of this PR was chosen as a starting point and to keep runs around 30-40 minutes.

TEST PLAN:
runs on remote push

### CODE REVIEW

File: neuralmagic/tests/skip-for-remote-push.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+tests/kernels/test_moe.py
+tests/kernels/test_layernorm.py
+tests/kernels/test_attention.py
+tests/distributed/test_custom_all_reduce.py
+tests/distributed/test_comm_ops.py
+tests/prefix_caching/test_prefix_caching.py
+tests/models/test_mistral.py
+tests/models/test_compressed_memory.py
+tests/models/test_marlin.py
+tests/models/test_compressed.py
```
Comment (mgoin) [inline neuralmagic/tests/skip-for-remote-push.txt:14]: I think we should keep this sparsity correctness test around

**Review by mgoin — APPROVED** [review]

File: .github/actions/nm-code-coverage/action.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: test nm-vllm
+description: "test nm-vllm via, 'pytest tests/'"
```
Comment (varun-sundar-rabindranath) [inline .github/actions/nm-code-coverage/action.yml:2]: cruft name

**Review by varun-sundar-rabindranath — APPROVED** [review]

File: .github/actions/nm-code-coverage/action.yml (hunk 2 above)
Comment (andy-neuma) [reply .github/actions/nm-code-coverage/action.yml:2]: thanks, i'll update before merging

File: neuralmagic/tests/skip-for-remote-push.txt (hunk 1 above)
Comment (andy-neuma) [reply neuralmagic/tests/skip-for-remote-push.txt:14]: cool.

**Pull request merged**